### ISSUE TITLE
Segmentation fault using --psm 0

### ISSUE THREAD

**Issue by konstantin-dzreev** [tesseract-ocr/tesseract#1576]
### Environment

* **Tesseract Version**: tesseract 4.0.0-beta.1
* **Commit Number**: 45a65463247cfc6856a186341c6918da0b1f44be
* **Platform**: ubuntu 16.04

### Current Behavior:
``` 
$ TESSDATA_PREFIX=./bin/tesseract-4.0.0/tessdata LD_LIBRARY_PATH=./bin/tesseract-4.0.0/lib ./bin/tesseract-4.0.0/tesseract ../5.png stdout --psm 0

Warning. Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 345
[1]    15945 segmentation fault (core dumped)  TESSDATA_PREFIX=./bin/tesseract-4.0.0/tessdata LD_LIBRARY_PATH=  ../5.png   0
```

There are no issues with, lets say, 'psm 1':
```
$ TESSDATA_PREFIX=./bin/tesseract-4.0.0/tessdata LD_LIBRARY_PATH='./bin/tesseract-4.0.0/lib' ./bin/tesseract-4.0.0/tesseract ../5.png stdout --psm 1

Warning. Invalid resolution 0 dpi. Using 70 instead.
Estimating resolution as 345
<Extracted text goes here....>
```

**Comment by stweil** [comment]
It looks like you have built your own Tesseract binaries. Then I suggest that you use a debugger like gdb to get more information which helps to find the bug causing that crash.

If that is not possible, you should at least provide all information which is needed for others to reproduce the problem: exact version of the code, where did you get eng.traineddata from, if the bug only occurs with some image it is needed, too.

**Comment by konstantin-dzreev** [comment]
My bad. I tried datafiles coming from:

- https://github.com/tesseract-ocr/tessdata_best
- https://github.com/tesseract-ocr/tessdata_fast

and both did not work. I just noticed a note saying that they wont work with modes 0 and 2. The one from  https://github.com/tesseract-ocr/tessdata works fine. I'm closing the issue.

Thanks for quick response!

**Comment by stweil** [comment]
Any crash is something which needs to be fixed. Wrong configurations must result in an error message, not in a crash, so this is still an open bug.

**Comment by konstantin-dzreev** [comment]
Then I'm re-opening it...

**Comment by konstantin-dzreev** [comment]
I just re-read the doc again. It says that  ```Tesseract's oem modes '0' and '2' won't work with them```. Should ```psm``` mode ```'0'``` still work with xxx_best and xxx_fast trained datafiles? The trained files fail in ```psm '0'```, not ```oem '0'``` for me (```psm 2``` works fine).

**Comment by amitdo** [comment]
Seems like a duplicate of #1167.

**Comment by stweil** [comment]
I think this was fixed in commit 27ce472666a6332f38888cd6855e1162fb82c729. konstantin-dzreev, please test and close the issue if it was fixed.

**Comment by konstantin-dzreev** [comment]
It seems to work, thanks!
